### DISCUSSION TITLE
How to put the query result in a zustand store?

### DISCUSSION THREAD

**Discussion by amirhhashemi** [TanStack/query#2550] [Q&A] [answered]
I want to put the result of a query in a zustand store:
```
export const useAuthStore = create(() => ({
  authUser:  useAuthUserQuery()
  login: useLoginMutation().mutateAsync
}));
```
I can access the auth user everywhere using just `useAuthUserQuery` But I want to keep everything related to authentication in a single place and zustand might be a good choice. 

But as you might guess that gives me an `Invalid hook call. Hooks can only be called inside of the body of a function component` error because I called a hook outside of a function component. this is not the case when using context API. 

How should i handle this? Any opinion about this strategy?

**Comment by TkDodo** [comment, marked as answer]
I answered on stackoverflow: https://stackoverflow.com/questions/68690221/how-to-use-zustand-to-store-the-result-of-a-query
